globals.d.ts

```diff
@@ -52,6 +52,137 @@ export interface DatabaseObject {
   [key: string]: unknown;
 }
 
+/**
+ * Search result object; similar to `DatabaseObject` but without some properties.
+ */
+export type SearchResultsObject =
+  | SearchResultsObjectProps
+  | SearchResultsObjectGenerics;
+
+interface SearchResultsObjectProps {
+  "@id": string;
+  "@type": string[];
+}
+
+interface SearchResultsObjectGenerics {
+  [key: string]: unknown;
+}
+
+/**
+ * The `properties` object of a schema.
+ */
+export type SchemaProperties = SchemaPropertiesProps | SchemaPropertiesGenerics;
+
+export interface SchemaPropertiesProps {
+  title: string;
+  type: string;
+}
+
+export interface SchemaPropertiesGenerics {
+  [key: string]: unknown;
+}
+
+/**
+ * Describes schema objects.
+ */
+export interface Schema {
```

Comment:
It's also occurring to me that these are all JSON-schema and JSON-LD. Especially for `properties` field, I wonder if there are typescript types for JSON-schema already made?